File: c#/src/Docmost.Sdk/Services/CommentsApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class CommentsApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> CreateCommentAsync(CreateCommentDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("comments/create", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> FindPageCommentsAsync(FindPageCommentsRequest request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("comments", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetCommentAsync(CommentIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("comments/info", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateCommentAsync(UpdateCommentDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("comments/update", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> DeleteCommentAsync(CommentIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("comments/delete", request, cancellationToken).ConfigureAwait(false);
    }

}

File: typescript/src/services/comments-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { CommentIdDto, CreateCommentDto, FindPageCommentsRequest, UpdateCommentDto } from '../models/index.js';


export class CommentsApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async createComment(request: CreateCommentDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`comments/create`, request);
  }

  async findPageComments(request: FindPageCommentsRequest): Promise<ApiResponse<unknown>> {
    return this.http.post(`comments`, request);
  }

  async getComment(request: CommentIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`comments/info`, request);
  }

  async updateComment(request: UpdateCommentDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`comments/update`, request);
  }

  async deleteComment(request: CommentIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`comments/delete`, request);
  }

}
